internal/compiler-bridge/src/main/scala/xsbt/LocalToNonLocalClass.scala

```diff
@@ -30,15 +31,15 @@ import collection.mutable.Map
  *
  * Additionally, you can query whether a given class is local. Check `isLocal`'s documentation.
  */
-class LocalToNonLocalClass[G <: CallbackGlobal](val global: G) {
+class LocalToNonLocalClass[G <: Global](val global: G) {
   import global._
   private val cache: Map[Symbol, Symbol] = perRunCaches.newMap()
 
   def resolveNonLocal(s: Symbol): Symbol = {
-    assert(
-      phase.id <= sbtDependency.ownPhase.id,
-      s"Tried to resolve ${s.fullName} to a  non local classes but the resolution works up to sbtDependency phase. We're at ${phase.name}"
-    )
+//    assert(
```

Comment:
I suspect we need to be creative here, instead of commenting this code.

Reply:
absolutely.
Seems like one solution can be to split `LocalToNonLocalClass` to `CallbackLocalToNonLocalClass` which handles the zinc-specific logic (and knows of `CallbackGlobal `) and `InternalLocalToNonLocalClass` which handles the caching logic (and only knows of `Global`). `CallbackLocalToNonLocalClass` can be a wrapper for `InternalLocalToNonLocalClass`.

of-course maybe other refactoring options for the shared cache issues can be more elegant...